src/io/output/chkpt/CheckPointWriter.cpp

```diff
@@ -0,0 +1,39 @@
+#include "CheckPointWriter.h"
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+#include "io/xml_schemas/xsd_type_adaptors/InternalToXSDTypeAdapter.h"
+
+void CheckPointWriter::writeFile(const SimulationParams& params, size_t iteration,
+                                 const std::unique_ptr<ParticleContainer>& particle_container) const {
+    std::vector<Particle> particles{particle_container->begin(), particle_container->end()};
```

Comment:
Maybe not ideal to copy the full particle vector. A `getParticles()` function in the ParticleContainer would be better (with return type: `const std::vector<Particle>&`.

Also, how can this work if the other method is annotated with [[nodiscard]] but the return value is not used here?